scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -160,41 +188,112 @@
+                                 ; This is an exception. The writeback to datomic has occurred, so there's nothing to do except to delete the task from kubernetes
+                                 ; and remove it from our tracking.
+                                 (delete-task api-client pod)
+                                 [:expected/running :pod/failed] (pod-has-just-completed kcc existing-state-dict) ; TODO: May need to mark mea culpa retry
+                                 [:expected/running :pod/waiting]
+                                 (do ; This case is weird.
+                                   (kill-task-weird api-client expected-state-dict pod) ; This breaks our rule of calling pod-has-completed on a non-terminal pod state.
+                                   (pod-has-just-completed kcc existing-state-dict)) ; TODO: Should mark mea culpa retry
+
+                                 [:expected/completed :pod/failed] (delete-task api-client pod)
                                  [:expected/completed :missing] nil ; Cause it to be deleted.
                                  [:expected/killed :pod/waiting] (kill-task api-client expected-state-dict pod)
                                  [:expected/killed :pod/running] (kill-task api-client expected-state-dict pod)
 
-                                 [:expected/killed :pod/succeeded] (do
-                                                                     (pod-has-just-completed kcc existing-state-dict)
-                                                                     (remove-finalization-if-set-and-delete api-client expected-state-dict pod))
-                                 [:expected/killed :pod/failed]
-                                 (do
-                                   (pod-has-just-completed kcc existing-state-dict)
-                                   (remove-finalization-if-set-and-delete api-client expected-state-dict pod))
-                                 [:expected/killed :missing] (pod-was-killed kcc pod-name)
-                                 ; TODO: Implement :pod/unknown cases.
-                                 ; TODO: Implement :pod/pending cases.
-                                 ; TODO: Implement :pod/need-to-fail cases.
-                                 [:missing :pod/running] (kill-task api-client expected-state-dict pod)
-                                 [:missing :pod/pending] (kill-task api-client expected-state-dict pod)
-                                 [:missing :pod/need-to-fail] (kill-task api-client expected-state-dict pod)
-                                 [:missing :pod/succeeded] (remove-finalization-if-set-and-delete api-client expected-state-dict pod)
-                                 [:missing :pod/failed] (remove-finalization-if-set-and-delete api-client expected-state-dict pod)
+                                 [:expected/killed :pod/succeeded] ; There was a race and it completed normally before being it was killed.
```

Comment:
You can introduce a `do` to fit the comment. For consistency, can you begin the `do` next to the vector on the same line.

Reply:
Fine.

Reply:
not done yet?